src/mds/MDCache.h

```diff
@@ -68,10 +69,6 @@ class MMDSFragmentNotify;
 
 class ESubtreeMap;
 
-struct MDRequestImpl;
-typedef ceph::shared_ptr<MDRequestImpl> MDRequestRef;
-struct MDSlaveUpdate;
```

Comment:
Unfortunately no, because the intrusive_ptr needs access to the refcount in the parent class. :(